File: application/src/test/java/com/study/platform/support/fake/FakeTeamScheduleQueryPort.java
package com.study.platform.support.fake;

import com.study.platform.domain.team.dto.response.TeamScheduleResponse;
import com.study.platform.domain.team.port.TeamScheduleQueryPort;

import java.util.List;
import java.util.UUID;

public class FakeTeamScheduleQueryPort implements TeamScheduleQueryPort {

    private final FakeTeamScheduleRepository repository;

    public FakeTeamScheduleQueryPort(FakeTeamScheduleRepository repository) {
        this.repository = repository;
    }

    @Override
    public List<TeamScheduleResponse> findAllByTeamId(UUID teamId) {
        return repository.findAllByTeamId(teamId).stream()
                .map(TeamScheduleResponse::from)
                .toList();
    }
}

File: application/src/test/kotlin/com/study/platform/support/fake/FakeTeamScheduleQueryPort.kt
package com.study.platform.support.fake

import com.study.platform.domain.team.dto.response.TeamScheduleResponse
import com.study.platform.domain.team.port.TeamScheduleQueryPort
import java.util.UUID

class FakeTeamScheduleQueryPort(
    private val repository: FakeTeamScheduleRepository
) : TeamScheduleQueryPort {

    override fun findAllByTeamId(teamId: UUID): List<TeamScheduleResponse> =
        repository.findAllByTeamId(teamId).map { TeamScheduleResponse.from(it) }
}
